### PR TITLE
GTK3 + Cmake

### PR DESCRIPTION
**Pull request by JelleBootsma**
Includes:

- GTK3 update (Scroom and cpp plugin)
- GTK3 ruler
- CPP CMake update
- CPP Github actions CI

### CODE REVIEW

**Comment by JelleBootsma** [comment]
GTK 3 and CMake have been merged into 1 pull request, as the Scroom submodule was changed in both branches, and thus had to be merged first, in order to prevent merge conflicts down the line

File: .github/workflows/cmake.yml [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+#        # Execute tests defined by the CMake configuration.
+#        # See https://cmake.org/cmake/help/latest/manual/ctest.1.html for more detail
+#        run: xvfb-run ctest -C ${{env.BUILD_TYPE}}
+
+  build-windows:
+    runs-on: windows-latest
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
```
Comment (kees-jan) [inline .github/workflows/cmake.yml:95]: You guys haven't yet fixed `fetch-depth`

File: .gitmodules [hunk 2]
```diff
@@ -1,3 +1,3 @@
 [submodule "scroom"]
 	path = scroom
-	url = https://github.com/kees-jan/scroom.git
+	url = https://github.com/Group8Sep/scroom.git
```
Comment (kees-jan) [inline .gitmodules:3]: At the end of the project, this should point to the original repository. Accepting for now...

File: plugins/sep/CMakeLists.txt [hunk 3]
```diff
@@ -27,7 +27,8 @@ target_link_libraries(
   spsep
         PRIVATE project_options project_warnings
         PUBLIC PkgConfig::gtk threadpool PkgConfig::cairo
-        boost_filesystem
+        Boost::filesystem
+		tiledbitmap
```
Comment (kees-jan) [inline plugins/sep/CMakeLists.txt:31]: In this a tabs/spaces issue?

**Review by kees-jan — CHANGES_REQUESTED** [review]
* Please have a look at the static analysis warnings
* Please have a look at the compiler warnings that you've added
* Please make sure that the changes to Scroom also get merged

File: .github/workflows/cmake.yml (hunk 1 above)
Comment (JelleBootsma) [reply .github/workflows/cmake.yml]: Resolved in commit c0de167

File: plugins/sep/CMakeLists.txt (hunk 3 above)
Comment (JelleBootsma) [reply plugins/sep/CMakeLists.txt]: Yes it was,
Resolved in commit c0de167

**Pull request merged**